ui/app/adapters/sync/destination.js

```diff
@@ -4,32 +4,31 @@
  */
 
 import ApplicationAdapter from 'vault/adapters/application';
+import { pluralize } from 'ember-inflector';
 
 export default class SyncDestinationAdapter extends ApplicationAdapter {
-  namespace = 'v1';
+  namespace = 'v1/sys';
 
-  _baseUrl() {
-    return `${this.buildURL()}/sys`;
+  pathForType(modelName) {
+    return modelName === 'sync/destination' ? pluralize(modelName) : modelName;
   }
 
-  createRecord(store, type, snapshot) {
-    const { name, type: destinationType } = snapshot.attributes();
-    const url = `${this._baseUrl()}/sync/destinations/${destinationType}/${name}`;
-
-    return this.ajax(url, 'POST', { data: snapshot.serialize() }).then((resp) => ({
-      id: `${destinationType}/${name}`,
-      ...resp,
-    }));
+  urlForCreateRecord(modelName, snapshot) {
+    const { name } = snapshot.attributes();
+    return `${super.urlForCreateRecord(modelName, snapshot)}/${name}`;
   }
 
-  // modelName is sync/destinations/:type
-  // id is the destination name
-  urlForFindRecord(id, modelName) {
-    return `${this._baseUrl()}/${modelName}/${id}`;
+  urlForDeleteRecord(id, modelName, snapshot) {
+    const { name, type } = snapshot.attributes();
+    // the modelName may be sync/destination or a child depending if it was initiated from the list or details view
+    // since the id for sync/destinations is type/name it will actually generate the correct url but the slash will be encoded
+    // if we normalize to use the child model name for url generation instead things will be consistent
+    const normalizedModelName =
+      modelName === 'sync/destination' ? `${pluralize(modelName)}/${type}` : modelName;
+    return `${super.urlForDeleteRecord(name, normalizedModelName, snapshot)}`;
   }
```

Comment:
The comment sums it up but I noticed this while using mirage since it was only matching the path on `/` and not `%2F`.